### ISSUE TITLE
TIDALARCH

### ISSUE THREAD

**Issue by TSAMBALI** [TSAMBALI/Ariadne_Tsambali#42]
# TidalArch: Navigating the Silent History
### 1. Abstract
The "TidalArch" project represents a paradigm shift in underwater archaeology, bridging the technological gap between advanced Generative AI and historical preservation. By leveraging the **Gemini 2.5 and 3 series** models, we provide marine archaeologists with a real-time, multimodal assistant capable of identifying, verifying, and reconstructively visualizing artifacts in high-pressure, low-visibility environments. This report details the inspiration, development, and mathematical foundations of the TidalArch platform, submitted for the TIDALHACK:26 Spring Hackathon.

---

### 2. Inspiration: The Ocean's Forgotten Libraries
The inspiration for TidalArch stems from a sobering reality: humanity knows more about the surface of the Moon than about the floor of our own oceans. Over **95% of the Earth's oceans** remain unexplored. Submerged within this "Great Unknown" are millions of shipwrecks, sunken cities, and prehistoric settlements—true "libraries" of human history currently subject to rapid degradation due to climate change, ocean acidification, and treasure looting.

Traditional marine archaeology is a painstakingly slow and expensive process. During our research, we identified three critical bottlenecks:
1.  **Life Support Constraints**: Divers are governed by the physics of partial pressures. Every second spent attempting to identify an artifact is a second of oxygen consumed.
2.  **Cognitive Load**: In low-visibility environments (silt, organic "marine snow"), the human eye often fails to recognize geometric patterns that indicate man-made artifacts.
3.  **The Information Gap**: Ground-truth historical data is often stored in terrestrial databases. A diver cannot easily "Google" a pottery shard while 30 meters deep.

Our vision was to create a **"Digital Dive Buddy"**—an expert intelligence that lives in the diver’s ear, seeing what they see, hearing what they say, and instantly grounding those observations in the sum of human historical knowledge.

---

### 3. The Interdisciplinary Approach: AI Meets Marine Science
The TidalHack prompt challenged us to apply AI to fields that have historically underutilized these technologies. Marine Archaeology is the perfect candidate. It requires a synthesis of **Physics (Acoustics)**, **History (Archaeology)**, and **Data Science (Computer Vision/NLP)**.

#### 3.1. Mathematical Foundations of Underwater Communication
To facilitate real-time voice assistance, we had to account for the physics of underwater communication. Sound behaves fundamentally differently in water than in air. The speed of sound $c$ in seawater is not a constant; it is a dynamic variable influenced by temperature, salinity, and pressure. 

We utilize the Mackenzie equation for high-precision mission planning:
$$c(T, S, z) = 1448.96 + 4.591T - 5.304 \times 10^{-2}T^2 + 2.374 \times 10^{-4}T^3 + 1.340(S - 35) + 1.630 \times 10^{-2}z + ...$$
*Where $T$ is temperature ($^\circ C$), $S$ is salinity (ppt), and $z$ is depth (m).*

Furthermore, the signal-to-noise ratio ($SNR$) for a diver’s voice is impacted by the absorption coefficient $\alpha$:
$$TL = 20 \log_{10}(R) + \alpha R \times 10^{-3}$$
Where $TL$ is the transmission loss and $R$ is the range. Our AI models are fine-tuned to understand audio inputs that have undergone significant attenuation and spectral shifting due to the high-pressure environment of the diver's helmet.

---

### 4. How We Built TidalArch: The Technical Odyssey
The architecture of TidalArch is a distributed React system designed for low-latency, high-reliability interaction.

#### 4.1. The Gemini Multimodal Stack
We utilized the full spectrum of the Google GenAI SDK to create a cohesive ecosystem:
-   **Gemini 3 Pro (The Command Center)**: Used for complex reasoning and historical cross-referencing. When a diver describes an artifact, the Pro model performs deep-dive analysis.
-   **Gemini 3 Flash (The Real-Time Engine)**: Powering the "Artifact Lab," this model uses **Google Search Grounding** to ensure that every identification is backed by real-world maritime registries (e.g., the UNESCO shipwreck database).
-   **Gemini 2.5 Flash Native Audio (The Bio-Link)**: This is our flagship feature. It enables a raw PCM audio stream that facilitates a continuous conversation. We avoided standard file-based APIs to achieve the lowest possible latency.
-   **Gemini 2.5 Flash Image (The Visualizer)**: This allows "Restorative Visualization." By prompting the model with an image of a corroded artifact, we can generate a "hypothetical restoration" to help archaeologists visualize the original state.

#### 4.2. Audio Engineering and PCM Processing
Implementation of the Live Assist required custom audio handlers. Since the Live API sends raw PCM bytes, we had to implement a manual decoding pipe:
```typescript
async function decodeAudioData(data: Uint8Array, ctx: AudioContext) {
  const dataInt16 = new Int16Array(data.buffer);
  const buffer = ctx.createBuffer(1, dataInt16.length, 24000);
  const channelData = buffer.getChannelData(0);
  for (let i = 0; i < dataInt16.length; i++) {
    channelData[i] = dataInt16[i] / 32768.0;
  }
  return buffer;
}
```
This ensures that the AI's "voice" reaches the diver without the overhead of container parsing (like .wav or .mp3), which is crucial for safety-critical real-time feedback.

---

### 5. Learning and Growth: What the Hackathon Taught Us
Building TidalArch in a single day was an exercise in "Aggressive Learning."

**1. The Complexity of Grounding**: We initially believed AI could simply "know" history. We quickly learned that in archaeology, "hallucinations" are not just annoying; they are a threat to heritage. If an AI falsely identifies a common stone as a rare artifact, it wastes mission resources. We learned to strictly enforce **Search Grounding**, requiring the model to cite sources for every identification.

**2. Physicality of Data**: We learned how significantly environment impacts data. Underwater images are often "blue-shifted." We had to learn how to prompt Gemini to "compensate for the water column" when analyzing photos, essentially teaching the AI to "squint" through the silt.

**3. Safety-First AI**: We realized that in high-stakes fields like diving, AI cannot just be a "chatbot." It must be a "guardian." This led to the implementation of the proactive safety warning system in the Live Assist view—reminding divers of their $O_2$ levels and No Decompression Limits ($NDL$).

---

### 6. Challenges: The Hurdles of the Deep
Our journey was not without its "bends."

#### 6.1. The Latency Gap
The Gemini Live API is incredibly fast, but maintaining a synchronized stream of image frames and high-fidelity audio over a simulated satellite link (typical for research vessels) created significant jitter. We solved this by implementing a **Jitter Buffer**—a logic gate that schedules audio chunks to play at precise intervals:
$$StartTime_{next} = \max(Time_{current}, StartTime_{prev} + Duration_{prev})$$
This ensures that even if the network fluctuates, the AI's instructions remain intelligible.

#### 6.2. The "Helium Speech" Problem
In deep dives using Heliox (Helium-Oxygen) mixes, the speed of sound inside the helmet increases ($c_{Helium} \approx 970 m/s$ vs $c_{Air} \approx 340 m/s$), causing the diver's voice to sound high-pitched. While we couldn't build a full DSP pitch-shifter during the hackathon, we learned that Gemini's native audio model is surprisingly robust at identifying speech patterns even when the formants are shifted, a testament to the "Native Audio" training of the 2.5 series.

#### 6.3. Grounding Niche Data
Maritime history is often documented in obscure, non-digitized archives. While Google Search is powerful, it sometimes missed specific shipwreck coordinates. We had to implement a "Confidence Threshold" UI component—if the AI's grounding metadata returns few results, the system warns the archaeologist that the finding is "Speculative."

---

### 7. Future Vision: The Next Wave
TidalArch is just the surface. In the next iteration, we plan to integrate:
-   **Veo 3.1 Integration**: Using the start-frame and end-frame capabilities to generate 10-second videos of "The Sinking Event," helping historians reconstruct the final moments of a vessel.
-   **Haptic Feedback**: Connecting the AI to the diver's suit, where the AI can "pulse" the diver's left arm if it detects a significant artifact in their peripheral vision.
-   **Swarm Intelligence**: Coordinating multiple ROVs (Remotely Operated Vehicles) through a single Gemini Command Center, allowing for autonomous mapping of entire underwater cities.

---

### 8. Conclusion: A Call to Action
The TidalHack:26 theme—*Inspiring the next wave of innovative AI powered applications*—is a reminder that the most impactful AI will not be built for "Big Tech," but for the "Small Fields" that hold our biggest secrets. Marine archaeology is a field of silence. By giving a voice to the silent history of our oceans, TidalArch ensures that the stories of those who sailed before us are not lost to the currents of time.

This project proves that when we combine the rigorous physics of marine science with the generative power of Gemini, we can create tools that don't just "process data," but actually "protect history."

*Built with passion, caffeine, and the spirit of exploration.*
*TIDALHACK:26 - Team TidalArch.*

**************PLEASE REFER ON THE NEXT LINK FOR PART OF MY CODING*******************AS:- https://ai.studio/apps/drive/13qK3Ho8mTmV9LlryHzziEDCQH3aMZAeu?fullscreenApplet=true

****************FOR MORE DETAILS REFER TO MY ATTACHED FILES FOR ALL *.PY CODING***************

BLESSINGS. THANK YOU

[TYPE.PY](https://github.com/user-attachments/files/25153836/TYPE.PY)
[GEMINIs.PY](https://github.com/user-attachments/files/25153842/GEMINIs.PY)
[METADATA.PY](https://github.com/user-attachments/files/25153837/METADATA.PY)
[INDEX.PY](https://github.com/user-attachments/files/25153838/INDEX.PY)
[INDEX.HTML](https://github.com/user-attachments/files/25153834/INDEX.HTML)
[SIDEBAR.PY](https://github.com/user-attachments/files/25153835/SIDEBAR.PY)
[LIVEASSISTANT.PY](https://github.com/user-attachments/files/25153841/LIVEASSISTANT.PY)
[DASHBOARD.PY](https://github.com/user-attachments/files/25153840/DASHBOARD.PY)
[ARTIFACT.PY](https://github.com/user-attachments/files/25153833/ARTIFACT.PY)
[APP.PY](https://github.com/user-attachments/files/25153839/APP.PY)
